### DISCUSSION TITLE
Questions about plugins and the beets DB

### DISCUSSION THREAD

**Discussion by phirestalker** [beetbox/beets#4498] [Q&A] [answered]
I wrote a plugin to create par2 sets for each music file [here](https://github.com/phirestalker/beets-musicintegrity). As it is now I think it performs these actions:
1. creates par2 file for all new imports (only once)
2. checks a track with par2 before making changes to it by hooking on the write event (which I assume actually happens before a write).
3. if the check passes it allows the change and creates a new par2 set for the changed file.

Can anyone verify that it does indeed do these things correctly?

Also, while developing this plugin my library is now in a state where the file metadate is out of sync with the beets DB. This raises a few questions related to how I get out of this mess.

Does the DB store all tags that plugins add to a file (such as the acousticbrainz tags)?
If so, how can I force beets to write all of it back to the file tags thus generating a (hopefully) correct par2 set for the files?

Whew, I think that covers it.
Thanks, all.

**Comment by sampsyo** [comment, marked as answer]
The main commands for "synchronizing" the beets database with on-disk file metadata are `beet write` (database -> file tags) and `beet update` (file tags -> database).
